logging/logger_registry.go

```diff
@@ -37,17 +34,15 @@ func (lr *loggerRegistry) deregisterLogger(name string) bool {
 	return ok
 }
 
-func (lr *loggerRegistry) loggerNamed(name string) (logger Logger, ok bool) {
+func (lr *Registry) loggerNamed(name string) (logger Logger, ok bool) {
 	lr.mu.RLock()
 	defer lr.mu.RUnlock()
 	logger, ok = lr.loggers[name]
 	return
 }
 
-func (lr *loggerRegistry) updateLoggerLevelWithCfg(name string) error {
-	lr.mu.RLock()
-	defer lr.mu.RUnlock()
-
+// updateLoggerLevelWithCfg must be called with a read or write lock on `lr.mu`.
```

Comment:
Highlighting this. I also want to take an axe to the private API because these private functions are either:
* not called
* only called by other private functions

Making the locking requirements confusing.

But I'm saving that for some later refactor after the behavior changes/requirements are locked down.